blockchain/state/statedb.go

```diff
@@ -1102,14 +1102,13 @@ func (s *StateDB) PrepareAccessList(rules params.Rules, sender, feepayer, coinba
 	for _, addr := range precompiles {
 		s.AddAddressToAccessList(addr)
 	}
-	// Uncommented below code because we do not support optional accessList yet (written at eip-2930).
 	// Optional accessList is the accessList mentioned through tx args.
-	//for _, el := range list {
-	//	s.AddAddressToAccessList(el.Address)
-	//	for _, key := range el.StorageKeys {
-	//		s.AddSlotToAccessList(el.Address, key)
-	//	}
-	//}
+	for _, el := range list {
+		s.AddAddressToAccessList(el.Address)
+		for _, key := range el.StorageKeys {
+			s.AddSlotToAccessList(el.Address, key)
+		}
+	}
```

Comment:
Can we check `rules.IsCancun` here instead of state_transition.go:346?
The similar `rules.IsShanghai` is also below here.

Reply:
Good point, thanks.

BTW, should we move Kore check in `state_transition.go` to `statedb.go` too? It looks like geth does so.

Reply:
That's not bad. Please check other instances of `if IsKore { PrepareAccessList }` pattern.